src/Controls/src/Core/Handlers/Items/CarouselViewHandler.Android.cs

```diff
@@ -1,25 +1,79 @@
+
+namespace Microsoft.Maui.Controls.Handlers.Items
 {
 	public partial class CarouselViewHandler : ItemsViewHandler<CarouselView>
 	{
+		double _widthConstraint;
+		double _heightConstraint;
+
 		protected override IItemsLayout GetItemsLayout() => VirtualView.ItemsLayout;
 
-		[MissingMapper]
-		public static void MapCurrentItem(CarouselViewHandler handler, CarouselView carouselView) { }
+		protected override ItemsViewAdapter<CarouselView, IItemsViewSource> CreateAdapter()
+		{
+			return new CarouselViewAdapter<CarouselView, IItemsViewSource>(VirtualView, (view, context) => new SizedItemContentView(Context, GetItemWidth, GetItemHeight));
+		}
+		protected override RecyclerView CreatePlatformView()
+		{
+			return new MauiCarouselRecyclerView(Context, GetItemsLayout, CreateAdapter);
+		}
+
+		public static void MapIsSwipeEnabled(CarouselViewHandler handler, CarouselView carouselView)
+		{
+			(handler.PlatformView as IMauiCarouselRecyclerView).IsSwipeEnabled = carouselView.IsSwipeEnabled;
+		}
+
+		public static void MapIsBounceEnabled(CarouselViewHandler handler, CarouselView carouselView)
+		{
+			handler.PlatformView.OverScrollMode = carouselView?.IsBounceEnabled == true ? OverScrollMode.Always : OverScrollMode.Never;
+		}
+
+		public static void MapPeekAreaInsets(CarouselViewHandler handler, CarouselView carouselView)
+		{
+			(handler.PlatformView as IMauiRecyclerView<CarouselView>).UpdateAdapter();
+		}
+
+		public static void MapPosition(CarouselViewHandler handler, CarouselView carouselView)
+		{
+			(handler.PlatformView as IMauiCarouselRecyclerView).UpdateFromPosition();
+		}
+
+		public static void MapCurrentItem(CarouselViewHandler handler, CarouselView carouselView)
+		{
+			(handler.PlatformView as IMauiCarouselRecyclerView).UpdateFromCurrentItem();
+		}
+
+		public override Size GetDesiredSize(double widthConstraint, double heightConstraint)
+		{
+			_widthConstraint = widthConstraint;
+			_heightConstraint = heightConstraint;
+			return base.GetDesiredSize(widthConstraint, heightConstraint);
+		}
+
+		int GetItemWidth()
+		{
+			var itemWidth = (int)Context?.ToPixels(_widthConstraint);
 
-		[MissingMapper]
-		public static void MapPosition(CarouselViewHandler handler, CarouselView carouselView) { }
+			if ((PlatformView as IMauiRecyclerView<CarouselView>)?.ItemsLayout is LinearItemsLayout listItemsLayout && listItemsLayout.Orientation == ItemsLayoutOrientation.Horizontal)
+				itemWidth = (int)(PlatformView.Width - Context?.ToPixels(VirtualView.PeekAreaInsets.Left) - Context?.ToPixels(VirtualView.PeekAreaInsets.Right) - Context?.ToPixels(listItemsLayout.ItemSpacing));
 
-		[MissingMapper]
-		public static void MapIsBounceEnabled(CarouselViewHandler handler, CarouselView carouselView) { }
+			return itemWidth;
+		}
 
-		[MissingMapper]
-		public static void MapIsSwipeEnabled(CarouselViewHandler handler, CarouselView carouselView) { }
+		int GetItemHeight()
+		{
+			var itemHeight = (int)Context?.ToPixels(_heightConstraint);
```

Comment:
The way the measure code is processed here isn't quite correct. 

If the _heightContraints is infinite the values start getting really messed up